packages/core/__tests__/cli/custom-extensions.test.ts

```diff
@@ -165,5 +173,24 @@ describe('CLI: custom extensions', () => {
         expect(result.stderr).toBe('');
       }
     );
+
+    test.runIf(semver.gte(typescript.version, '5.0.0'))(
+      'declarations work with `allowImportingTsExtensions: true`',
```

Comment:
I think we decided in Discord that driving this emit behavior off of `allowImportingTsExtensions` was a bad idea?

Reply:
For me, I'm trying to address one thing at a time -- if I can find where to delete `.gts` from the imports, then I can later un-do any `allowImportingTsExtensions`-specific behavior

Reply:
Rather than typing it out again, I'm going to just paste a link to the conversation we had in which I said repeatedly that modifying imports likely isn't the path forward here 😉 

https://discord.com/channels/480462759797063690/491905849405472769/1166783585890615377

I believe you, I and @simonihmig all agreed there that a Glint-specific flag to allow including the `.gts` part in the emitted declarations made the most sense?

Reply:
I don't actually recall / observe any agreement -- only [a suggestion](https://discord.com/channels/480462759797063690/491905849405472769/1166826588965445682).

Which, I don't know why we'd need it other than "abundance of caution".

Reply:
> Which, I don't know why we'd need it other than "abundance of caution".

Simon found a real-world public example of where this would be a breaking change.

> I don't actually recall / observe any agreement

```
Simon Ihmig — 10/25/2023 9:51 PM
I'd rather introduce a new flag then tbh
NullVoxPopuli — 10/25/2023 9:55 PM
works for me / I have no problem with that / simpler = better
```

I read that as an agreement (and while I had `+1`ed an earlier comment, I'll make my agreement explicit here: I think that's the preferable approach)

Reply:
:100: